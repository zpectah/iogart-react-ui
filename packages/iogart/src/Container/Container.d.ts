import { ContainerProps } from './types';

declare const Container: ContainerProps;

export default Container;
