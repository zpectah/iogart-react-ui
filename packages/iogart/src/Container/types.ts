import { ReactNode } from 'react';

import { IogartCommonUiProps } from '@iogart-react-ui/types';
import { ContainerMaxWidthKeys, ContainerComponentKeys } from './enums';

export interface ContainerProps extends IogartCommonUiProps {
    children?: ReactNode;
    component?: keyof typeof ContainerComponentKeys;
    maxWidth?: keyof typeof ContainerMaxWidthKeys;
    fluid?: boolean;
}

export interface useContainerParameters extends ContainerProps {}
export interface useContainerReturn extends useContainerParameters {}