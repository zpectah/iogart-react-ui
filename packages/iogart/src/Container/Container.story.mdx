import { Meta, Story } from '@storybook/addon-docs';

import Container from './Container';
export const Template = (args) => <Container {...args} />;

<Meta title="Components/Layout" component={Container} />

# Container

`@iogart-react-ui/iogart/Container`

```javascript
import { Container } from 'iogart-react-ui/iogart';
```

<Story
  name="Container"
  args={{
    children: (
      <>
        <p>
          Interdum at, auctor mattis quis odio ornare vestibulum mauris aliquam, maximus in nisi
          enim curabitur. Cras dictum venenatis, quis in ex sem lacinia odio morbi ullamcorper nec
          venenatis, leo vitae imperdiet aliquet bibendum. Sem tortor, donec ligula nec commodo
          molestie mauris vivamus potenti, arcu odio bibendum donec. Ligula neque, aliquam
          sollicitudin integer nulla augue fusce scelerisque, libero hendrerit quis suspendisse.
          Convallis egestas, suscipit suspendisse suscipit dui vel nulla proin dui et tempus, vel
          arcu ac hendrerit urna nisl. Consectetur vivamus, aliquam nunc non nulla sapien bibendum
          nec sollicitudin, nulla a posuere tellus donec.
        </p>
        <p>
          Lectus quis condimentum, at nullam sem lobortis neque sapien semper molestie integer,
          mollis fusce vivamus ut venenatis. Vivamus suscipit, rhoncus aliquam et iaculis lorem
          fringilla aliquet ligula vel, integer felis scelerisque varius ornare. Donec ac nec,
          luctus consectetur porta sit amet sem odio mauris eu integer, fermentum dolor ut phasellus
          euismod. Sed porttitor ut, pulvinar tellus integer odio a congue lacinia, faucibus donec
          dui vel metus. fringilla dolor quis, metus a lorem vitae nec commodo euismod nunc
          curabitur interdum, aliquam porttitor phasellus lectus. Dui aliquet, ornare sed metus mi
          orci ante scelerisque rutrum, odio leo enim in libero lectus quis.
        </p>
      </>
    ),
  }}
>
  {Template.bind({})}
</Story>
