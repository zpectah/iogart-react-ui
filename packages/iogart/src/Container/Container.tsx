import React, { useMemo } from 'react';
import styled from '@emotion/styled';

import {
    breakpoints,
    container,
} from '@iogart-react-ui/styles';
import { ContainerProps } from './types';
import useContainer from './useContainer';

const Container = (props: ContainerProps) => {
    const {
        children,
        component = 'div',
        maxWidth,
        fluid,
        ...rest
    } = props;

    const updateProps = useContainer({
        ...rest
    });

    const renderElement = useMemo(() => {
        const isa_xs = (!maxWidth || maxWidth === 'xs');
        const isa_sm = (!maxWidth || maxWidth === 'sm');
        const isa_md = (!maxWidth || maxWidth === 'md');
        const isa_lg = (!maxWidth || maxWidth === 'lg');
        const isa_xl = (!maxWidth || maxWidth === 'xl');
        const isa_xxl = (!maxWidth || maxWidth === 'xxl');
        const bp_xs = `@media (min-width: ${breakpoints.xs.min}px) { max-width: 100%; }`;
        const bp_sm = `@media (min-width: ${breakpoints.sm.min}px) { max-width: ${container.sm}px; }`;
        const bp_md = fluid ? '' : `@media (min-width: ${breakpoints.md.min}px) { max-width: ${container.md}px; }`;
        const bp_lg = fluid ? '' : `@media (min-width: ${breakpoints.lg.min}px) { max-width: ${container.lg}px; }`;
        const bp_xl = fluid ? '' : `@media (min-width: ${breakpoints.xl.min}px) { max-width: ${container.xl}px; }`;
        const bp_xxl = fluid ? '' : `@media (min-width: ${breakpoints.xxl.min}px) { max-width: ${container.xxl}px; }`;

        const Element = styled(component)`
          max-width: 100%;
          margin-left: auto;
          margin-right: auto;
          ${isa_xs && bp_xs}
          ${isa_sm && bp_sm}
          ${isa_md && bp_md}
          ${isa_lg && bp_lg}
          ${isa_xl && bp_xl}
          ${isa_xxl && bp_xxl}                              
        `;

        return (
            <Element
                className="dpr-container"
                {...updateProps}
            >
                {children}
            </Element>
        );
    }, [ children, fluid, maxWidth, updateProps ]);

    return (
        <>
            {renderElement}
        </>
    );
};

export default Container;
