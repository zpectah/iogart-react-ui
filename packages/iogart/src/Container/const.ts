export const CONTAINER_CLASSNAME = 'Container';
export const CONTAINER_CONTENT_CLASSNAME = 'Container-content';
