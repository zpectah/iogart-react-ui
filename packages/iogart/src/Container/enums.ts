export enum ContainerElementTypeKeys {
  main = 'main',
  div = 'div',
  section = 'section',
  article = 'article',
}

export enum ContainerMaxWidthKeys {
  xs = 'xs',
  sm = 'sm',
  md = 'md',
  lg = 'lg',
  xl = 'xl',
  xxl = 'xxl',
}
