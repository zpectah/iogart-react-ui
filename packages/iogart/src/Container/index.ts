export { default as Container } from './Container';
export { default as useContainer } from './useContainer';
export * from './types';
