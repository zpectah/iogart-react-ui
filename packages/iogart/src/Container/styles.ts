import { createUseStyles } from 'react-jss';
import { withIogartThemeProps, THEME_BREAKPOINT_CONTAINER, THEME_BREAKPOINTS } from '@iogart-react-ui/styles';

interface useContainerStylesProps extends withIogartThemeProps {}

type useContainerStylesClassNames = 'root' | 'fluid' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';

const useContainerStyles = createUseStyles<useContainerStylesClassNames, useContainerStylesProps>({
  root: {
    maxWidth: '100%',
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingLeft: '.5rem',
    paddingRight: '.5rem',
  },
  fluid: {
    maxWidth: '100%',
  },
  xs: {
    [`@media (min-width: ${THEME_BREAKPOINTS.xs.min}px)`]: {
      maxWidth: '100%',
    },
  },
  sm: {
    [`@media (min-width: ${THEME_BREAKPOINTS.sm.min}px)`]: {
      maxWidth: THEME_BREAKPOINT_CONTAINER.sm,
    },
  },
  md: {
    [`@media (min-width: ${THEME_BREAKPOINTS.md.min}px)`]: {
      maxWidth: THEME_BREAKPOINT_CONTAINER.md,
    },
  },
  lg: {
    [`@media (min-width: ${THEME_BREAKPOINTS.lg.min}px)`]: {
      maxWidth: THEME_BREAKPOINT_CONTAINER.lg,
    },
  },
  xl: {
    [`@media (min-width: ${THEME_BREAKPOINTS.xl.min}px)`]: {
      maxWidth: THEME_BREAKPOINT_CONTAINER.xl,
    },
  },
  xxl: {
    [`@media (min-width: ${THEME_BREAKPOINTS.xxl.min}px)`]: {
      maxWidth: THEME_BREAKPOINT_CONTAINER.xxl,
    },
  },
});

export default useContainerStyles;
