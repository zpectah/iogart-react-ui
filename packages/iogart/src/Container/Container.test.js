import renderer from 'react-test-renderer';

import Container from './Container';

it('Container - common', () => {

    const component = renderer.create(
        <Container> <div>A</div><div>B</div><div>C</div> </Container>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
