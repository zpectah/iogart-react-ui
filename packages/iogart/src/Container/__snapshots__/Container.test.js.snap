// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container - common 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-eg7dfx{max-width:100%;margin-left:auto;margin-right:auto;}@media (min-width: 0px){.css-eg7dfx{max-width:100%;}}@media (min-width: 576px){.css-eg7dfx{max-width:540px;}}@media (min-width: 768px){.css-eg7dfx{max-width:720px;}}@media (min-width: 992px){.css-eg7dfx{max-width:960px;}}@media (min-width: 1200px){.css-eg7dfx{max-width:1140px;}}@media (min-width: 1400px){.css-eg7dfx{max-width:1320px;}}",
      }
    }
    data-emotion="css eg7dfx"
  />,
  <div
    className="iogart-container css-eg7dfx"
    style={{}}
  >
     
    <div>
      A
    </div>
    <div>
      B
    </div>
    <div>
      C
    </div>
     
  </div>,
]
`;
